Add channel ID to context cleared event (#2043)
finos/FDC3: website/docs/api/spec.md

- Provide [`ContextMetadata`](ref/Metadata#contextmetadata) (including `source` and `timestamp`) to [`ContextHandler`](ref/Types#contexthandler) functions registered via [`addContextListener`](ref/Channel#addcontextlistener).
- Return the most recent context (and associated metadata for [`getCurrentContextWithMetadata`](ref/Channel#getcurrentcontextwithmetadata)) matching the requested type, or `null` if no matching context is found.
- Return `null` from [`getCurrentContext`](ref/Channel#getcurrentcontext) and [`getCurrentContextWithMetadata`](ref/Channel#getcurrentcontextwithmetadata) for context types that have been cleared via [`clearContext`](ref/Channel#clearcontext), until new context of that type is broadcast.

> - Notify listeners registered for the `contextCleared` event (via [`addEventListener`](ref/Channel#addeventlistener)) when [`clearContext`](ref/Channel#clearcontext) is called, including the channel's `channelId` and the `contextType` that was cleared, or `null` if all context types were cleared. Desktop Agents MAY deliver only events for the individual channel on which the listener was registered, or MAY deliver events for all public channels.

Reviewer: ```suggestion
- Notify listeners registered for the `contextCleared` event (via [`addEventListener`](ref/Channel#addeventlistener)) when [`clearContext`](ref/Channel#clearcontext) is called, including the channel's `channelId` and the `contextType` that was cleared, or `null` if all context types were cleared. Desktop Agents MUST deliver only events for the individual channel on which the listener was registered.
```
